app/views/spree/admin/orders/index.html.haml

```diff
@@ -23,8 +23,14 @@
   = render partial: 'per_page_controls'
 
   - if Spree::Config[:enable_invoices?]
-    %button.invoices-modal{'ng-controller' => 'bulkInvoiceCtrl', 'ng-click' => 'createBulkInvoice()', 'ng-disabled' => 'selected_orders.length == 0'}
+    %button.invoices-modal{'ng-controller' => 'bulkInvoiceCtrl', 'ng-click' => 'createBulkInvoice()', 'ng-disabled' => 'selected_orders.length == 0 || !q.completed_at_not_null'}
```

Comment:
This changes live on the page when you toggle the "Only show complete orders" checkbox. So it's possible to just tick that box but not reload the page which means that you can still select an incomplete order and run into the same issue of the infinite spinner.